primitives/src/v2/mod.rs

```diff
@@ -1582,6 +1612,66 @@ pub fn supermajority_threshold(n: usize) -> usize {
 	n - byzantine_threshold(n)
 }
 
+/// Vector indexed via key value type e.g. ValidatorIndex
+#[derive(Clone, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(PartialEq))]
+pub struct TypeVec<K: From<usize>, V: Clone>(TiVec<K, V>);
+
+impl<K: From<usize>, V: Clone> TypeVec<K, V> {
+	///Creates new TypeVec object
+	pub fn new() -> Self {
+		TypeVec::<K, V> { 0: TiVec::<K, V>::new() }
+	}
+}
+
+impl<K: From<usize>, V: Clone> From<Vec<V>> for TypeVec<K, V> {
+	fn from(vec: Vec<V>) -> Self {
+		let mut type_vec: TypeVec<K, V> = TypeVec::new();
+		for value in vec.into_iter() {
+			type_vec.0.push(value)
+		}
+		type_vec
+	}
+}
+
+#[cfg(feature = "std")]
+impl<K: From<usize>, V: Clone> MallocSizeOf for TypeVec<K, V> {
+	fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+		0
```

Comment:
incorrect, should delegate to `Vec` impl

Reply:
Attempting this is causing issues.

Reply:
`#[cfg(feature = "std")]`
has to be implemented in numerous places

Reply:
Attempted implementation:
```
impl<K: From<usize>, V: Clone> MallocSizeOf for TypeVec<K, V> {
	fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
		self.to_vec().size_of(ops)
	}
}
```